### PR TITLE
Bug 1828516 - Hide Whats New button

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1828516

### CODE REVIEW

**Comment by jonalmeida** [comment]
Why are we changing the visibility of the button instead of removing it as the bug mentioned?

**Comment by sarah541** [comment]
> Why are we changing the visibility of the button instead of removing it as the bug mentioned?

I may have seen we could hide/ or remove in the bug's description. Moreover is it possible we might need to bring this back and would be easy to just un-hide it?

**Comment by jonalmeida** [comment]
> I may have seen we could hide/ or remove in the bug's description. Moreover is it possible we might need to bring this back and would be easy to just un-hide it?

@sarah541 we can just revert the change if we wanted to. We have git history. 🙂

File: focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -51,41 +42,6 @@ class SettingsFragment : BaseSettingsFragment() {
         return super.onPreferenceTreeClick(preference)
     }
 
-    override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
-        menuInflater.inflate(R.menu.menu_settings_main, menu)
-    }
-
-    override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
-        if (menuItem.itemId == R.id.menu_whats_new) {
```
Comment (jonalmeida) [inline focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:59]: This ID isn't needed any more.

File: focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -51,41 +42,6 @@ class SettingsFragment : BaseSettingsFragment() {
...
-        }
-        return false
-    }
-
-    private fun whatsNewClicked() {
-        val context = requireContext()
-
-        SettingsScreen.whatsNewTapped.add()
-
-        TelemetryWrapper.openWhatsNewEvent(WhatsNew.shouldHighlightWhatsNew(context))
```
Comment (jonalmeida) [inline focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:71]: The whole `whatsnew` package can go too.

File: focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt [hunk 3]
```diff
@@ -51,41 +42,6 @@ class SettingsFragment : BaseSettingsFragment() {
...
-
-        WhatsNew.userViewedWhatsNew(context)
-
-        val sumoTopic = if (AppConstants.isKlarBuild) {
-            SupportUtils.SumoTopic.WHATS_NEW_KLAR
-        } else {
-            SupportUtils.SumoTopic.WHATS_NEW_FOCUS
-        }
-
-        val url = SupportUtils.getSumoURLForTopic(SupportUtils.getAppVersion(context), sumoTopic)
```
Comment (jonalmeida) [inline focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:81]: This `getSumoURLForTopic` function can also be removed too.

File: focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt (hunk 2 above)
Comment (jonalmeida) [reply focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:71]: `TelemetryWrapper.openWhatsNewEvent` as well.

**Review by jonalmeida — CHANGES_REQUESTED** [review]
Continuing the review based on [jmahon's confirmation](https://bugzilla.mozilla.org/show_bug.cgi?id=1828516#c8) that we can continue to remove the code.

There are a few more things to remove that's left behind. The CI failures should also give some hints for the UI test changes that would be needed too.

File: focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt (hunk 3 above)
Comment (sarah541) [reply focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:81]: Why does this function need to be removed? It's being utilized at other places too.

Comment (jonalmeida) [reply focus-android/app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt:81]: Oops, you're right. 🙂

**Review by jonalmeida — APPROVED** [review]
Excellent, lgtm!

**Pull request merged**